tmp/test_instagram.rb

```diff
@@ -0,0 +1,166 @@
+require "sinatra"
+require "instagram"
+
+enable :sessions
+
+CALLBACK_URL = "http://localhost:9393/oauth/callback"
+
+Instagram.configure do |config|
+  config.client_id = "696296edc81f417ea3418708c35485dd"
+  config.client_secret = "76e99607d14f48dda453fb9c6109d55b"
+end
+
+get "/" do
+  '<a href="/oauth/connect">Connect with Instagram</a>'
+end
+
+get "/oauth/connect" do
+  redirect Instagram.authorize_url(:redirect_uri => CALLBACK_URL)
+end
+
+get "/oauth/callback" do
+  response = Instagram.get_access_token(params[:code], :redirect_uri => CALLBACK_URL)
+  session[:access_token] = response.access_token
+  redirect "/nav"
+end
+
+get "/nav" do
+  html =
+  """
+    <h1>Ruby Instagram Gem Sample Application</h1>
+    <ol>
+      <li><a href='/user_recent_media'>User Recent Media</a> Calls user_recent_media - Get a list of a user's most recent media</li>
+      <li><a href='/user_media_feed'>User Media Feed</a> Calls user_media_feed - Get the currently authenticated user's media feed uses pagination</li>
+      <li><a href='/location_recent_media'>Location Recent Media</a> Calls location_recent_media - Get a list of recent media at a given location, in this case, the Instagram office</li>
+      <li><a href='/media_search'>Media Search</a> Calls media_search - Get a list of media close to a given latitude and longitude</li>
+      <li><a href='/media_popular'>Popular Media</a> Calls media_popular - Get a list of the overall most popular media items</li>
+      <li><a href='/user_search'>User Search</a> Calls user_search - Search for users on instagram, by name or username</li>
+      <li><a href='/location_search'>Location Search</a> Calls location_search - Search for a location by lat/lng</li>
+      <li><a href='/location_search_4square'>Location Search - 4Square</a> Calls location_search - Search for a location by Fousquare ID (v2)</li>
+      <li><a href='/tags'>Tags</a>Search for tags, view tag info and get media by tag</li>
+      <li><a href='/limits'>View Rate Limit and Remaining API calls</a>View remaining and ratelimit info.</li>
+    </ol>
+  """
+  html
+end
+
+get "/user_recent_media" do
+  client = Instagram.client(:access_token => session[:access_token])
+  user = client.user
+  html = "<h1>#{user.username}'s recent media</h1>"
+  for media_item in client.user_recent_media
+    html << "<div style='float:left;'><img src='#{media_item.images.thumbnail.url}'><br/> <a href='/media_like/#{media_item.id}'>Like</a>  <a href='/media_unlike/#{media_item.id}'>Un-Like</a>  <br/>LikesCount=#{media_item.likes[:count]}</div>"
+  end
+  html
+end
+
+get '/media_like/:id' do
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.